### ISSUE TITLE
Pulling structure (TABL) with missing dependency

### ISSUE THREAD

**Issue by FreHu** [abapGit/abapGit#5294]
Repo setup:
- structure with field referring to data element
- data element

Initialize the repo and pull just the structure (I am simulating a missing dependency during a larger pull)

![image](https://user-images.githubusercontent.com/5097067/151344619-39fa6000-b2c0-4c6f-80d6-dfc8189590d3.png)

Then it says "no active version found", which is true but  the inactive object icon is missing. 
![image](https://user-images.githubusercontent.com/5097067/151344659-2c063131-e9e5-47db-8f25-25b58495dd90.png)

The object exists on the system in a partially active state:
![image](https://user-images.githubusercontent.com/5097067/151344816-d1d616b3-941d-43e7-98ae-2a694183bf9b.png)

Abapgit makes it look like the object will be created if you just pull, but if you keep repeating the pull, the DTEL will be pulled but you will not get out of the situation without activating the TABL manually.
![image](https://user-images.githubusercontent.com/5097067/151345066-3bc37bac-aeb4-46c5-8668-a4c8f73a298d.png)

There are a few improvements which can be made to this flow:
- show partly active objects like inactive objects
- Get rid of the duplicate message
- "no active version found" should point out that an inactive version was found and you need to activate it. Normally abapgit would activate or show an activation popup automatically
- the activation errors have the best information about the problem, but you will only see this popup once

**Comment by mbtools** [comment]
That's a bit messy indeed. 

1. Inactive check/icon

For partially active table, what status is returned in `ZCL_ABAPGIT_OBJECTS_SUPER->IS_ACTIVE`?

2. Duplicate message

TABL is in DDIC and LATE step. So it's trying twice to activate (since there might be missing dependencies in the first step). That's why we see the error twice. Solution might be a simple remove duplicates.

3. Activation errors

Those should be merged into the AG log. Needs to pass `ii_log` to `zcl_abapgit_objects_activation=>activate`.

**Comment by FreHu** [comment]
the is_active check is not reached because serialize throws due to missing `dd02v`
![image](https://user-images.githubusercontent.com/5097067/151356611-fec94700-23c1-45de-af59-f9ec883ee6ac.png)

**Comment by FreHu** [comment]
Based on the code in `zcl_abapgit_objects=>serialize`, this should be a general issue, the `is_active` logic will always be skipped if serialize throws.
```abap
 CREATE OBJECT li_xml TYPE zcl_abapgit_xml_output.

  ls_i18n_params-main_language         = iv_language.
  ls_i18n_params-main_language_only    = iv_main_language_only.
  ls_i18n_params-translation_languages = it_translation_langs.

  li_xml->i18n_params( ls_i18n_params ).

  li_obj->serialize( li_xml ).

  lo_files->add_xml( ii_xml      = li_xml
                     is_metadata = li_obj->get_metadata( ) ).

  rs_files_and_item-files = lo_files->get_files( ).

  check_duplicates( rs_files_and_item-files ).

  rs_files_and_item-item-inactive = boolc( li_obj->is_active( ) = abap_false ).
```

**Comment by mbtools** [comment]
good point. there's some try-catch needed there.

**Comment by FreHu** [comment]
I tried it locally but it doesn't help with this particular situation. 

```abap
  CALL FUNCTION 'RS_INACTIVE_OBJECTS_WARNING'
      EXPORTING
        suppress_protocol         = abap_false
        with_program_includes     = abap_false
        suppress_dictionary_check = abap_false
      TABLES
        p_e071                    = lt_e071_tadirs
        p_xmsg                    = lt_messages.

    IF lt_messages IS NOT INITIAL.
      ms_item-inactive = abap_true.
    ENDIF.
```
is not enough to identify the object as inactive.

Report `rutchkta` claims to identify partly active objects, but doesn't find mine either.
![image](https://user-images.githubusercontent.com/5097067/151387048-1c21bd66-128b-4a70-a3f7-c203c9c0a16d.png)

I will try to reproduce a similar situation where I pull a table which would require conversion, IIRC this also leaves the object in a partly active state and you have to go through SE11 first to proceed. It might give me some hints.

**Comment by FreHu** [comment]
This will detect inactive TABL, but it's getting too messy.
 
```abap
METHOD zif_abapgit_object~is_active.
    DATA: lv_name  TYPE ddobjname,
          ls_dd02v TYPE dd02v,
          ls_dd09l TYPE dd09l.
    lv_name = ms_item-obj_name.
    CALL FUNCTION 'DDIF_TABL_GET'
      EXPORTING
        name          = lv_name
        langu         = mv_language
        state         = 'A'
      IMPORTING
        dd02v_wa      = ls_dd02v
        dd09l_wa      = ls_dd09l
      EXCEPTIONS
        illegal_input = 1
        OTHERS        = 2.

    IF ls_dd02v IS INITIAL.
      CALL FUNCTION 'DDIF_TABL_GET'
        EXPORTING
          name          = lv_name
          langu         = mv_language
          state         = 'M'
        IMPORTING
          dd02v_wa      = ls_dd02v
          dd09l_wa      = ls_dd09l
        EXCEPTIONS
          illegal_input = 1
          OTHERS        = 2.
    ELSE.
      rv_active = abap_true.
    ENDIF.
  ENDMETHOD.
```
with the catch
```abap
TRY.
        li_obj->serialize( li_xml ).
      CATCH zcx_abapgit_exception INTO lx_serialize.
        rs_files_and_item-item-inactive = boolc( li_obj->is_active( ) = abap_false ).
        RETURN.
    ENDTRY.
```
But because my inactive object will serialize to zero files, the inactive flag is lost below and I still don't see the inactive object icon. Plus i'd like to avoid repeatedly calling the FM.

```abap
  METHOD add_to_return.

    FIELD-SYMBOLS: <ls_file>   LIKE LINE OF is_file_item-files,
                               <ls_return> LIKE LINE OF mt_files.

    LOOP AT is_file_item-files ASSIGNING <ls_file>.
    APPEND INITIAL LINE TO mt_files ASSIGNING <ls_return>.
      <ls_return>-file = <ls_file>.
      <ls_return>-file-path = iv_path.
      <ls_return>-item = is_file_item-item. " here is the inactive flag
    ENDLOOP.

  ENDMETHOD.
```

**Comment by mbtools** [comment]
Should be closed by #5302 + #5303. Could you please test them (together)?

If it works properly, I will adjust `DOMA`, `DTEL`, `ENQU`, `SHLP`, `TTYP`, `TYPE`, and `VIEW` accordingly.
